src/redux/reducer.ts

```diff
@@ -16,6 +24,12 @@ export default function reducer(
     case CHANGE_SECTION:
       return {
         sectionName: action.sectionName,
+        selectedItem: state.selectedItem,
+      };
+    case SELECT_ITEM:
+      return {
+        sectionName: state.sectionName,
+        selectedItem: action.item,
       };
```

Comment:
when we're returning the state inside the reducer, we should do:
```
return { ...state, selectedItem: action.item }
```
basically only explicity declare the keys that you're changing. that way when you add new fields to the state, you won't have to come back and add all those fields to each return as well